compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -32,7 +32,9 @@ object SourceLanguage:
       SourceLanguage.Java
     // Scala 2 methods don't have Inline set, except for the ones injected with `patchStdlibClass`
     // which are really Scala 3 methods.
-    else if denot.isClass && denot.is(Scala2x) || (denot.maybeOwner.lastKnownDenotation.is(Scala2x) && !denot.is(Inline)) then
+    else if denot.isClass && denot.is(Scala2x)
+          || (denot.maybeOwner.lastKnownDenotation.is(Scala2x) && !denot.is(Inline))
+          || denot.is(Param) && denot.maybeOwner.is(Method)  && denot.maybeOwner.maybeOwner.is(Scala2x) then
```

Comment:
```suggestion
          || denot.is(Param) && denot.maybeOwner.is(Method)  && denot.maybeOwner.maybeOwner.lastKnownDenotation.is(Scala2x) then
```
to be consistent with the method case?

We might also need to exclude the params of `Inline` methods? I don't know when that actually happens. What is even an `Inline` Scala 2 method?

Reply:
In my defense, this precedes my changes: https://github.com/scala/scala3/commit/6e58d25acd99ff44533730867cfc09873cb95f7a#diff-7cbbe7d1307696dd5d37afe4ce94c01f85f4945edb2c37b113245ad8c85264c1L30 .
I was only interested in making sure we are not recalculating the denotation, and likely missed the word nature of the check there. That is to say, I don't know what that is either :(